In [2]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k)*(factorial(n-k))))

#### 1) Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

##### a) Найти вероятность того, что все карты – крести:

Р = число исходов, благоприятствующих нашему событию / общее число исходов

число исходов, благоприятствующих нашему событию - это все карты крести:

In [3]:
combinations(13,4)*combinations(39,0)

715

общее число исходов:

In [4]:
combinations(52,4)

270725

In [6]:
P = combinations(13,4)*combinations(39,0) / combinations(52,4)
P

0.0026410564225690276

Ответ: 0.26%

##### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

Вариант 1

Найдем возможные комбинации с 1,2,3,4 тузами:

In [9]:
a = combinations(4,1)*combinations(48,3)
a

69184

In [10]:
b = combinations(4,2)*combinations(48,2)
b

6768

In [11]:
c = combinations(4,3)*combinations(48,1)
c

192

In [12]:
d = combinations(4,4)*combinations(48,0)
d

1

In [13]:
P = a + b + c + d
P

76145

Общее число исходов:

In [14]:
combinations(52,4)

270725

Вероятность:

In [15]:
P/combinations(52,4)

0.2812632745405855

Ответ: 28.13%

Вариант 2:

Будем действовать от обратного. вычтам из полной верояности вероятность того, что среди 4 карт не окажется ни одного туза:

In [17]:
1 - combinations(4,0)*combinations(48,4)/combinations(52,4)

0.2812632745405854

Ответ: 28.13%

#### 2) На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Найдем все возможные комбинации нажатия 3 кнопок из 10:

In [18]:
combinations(10,3)

120

Попытка 1, а значит число благоприятных исходов = 1

Найдем верояность:

In [19]:
1/combinations(10,3)

0.008333333333333333

Ответ: 0.8%

#### 3) В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

Речь идет о зависимых событиях. Вероятность того что первая деталь будет окрашена - 9/15, вторая деталь окрашена - 8/14, третья окрашена - 7/13

In [21]:
(9/15) * (8/14) * (7/13)

0.18461538461538457

Ответ: 18.46%

#### 4) В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Благоприятный исход: комбинации 2 билета из 2 выигрышные:

In [23]:
combinations(2,2)*combinations(98,0)

1

Общее число исходов:

In [24]:
combinations(100,2)

4950

вероятность что 2 из 100 - выигрышные:

In [25]:
combinations(2,2)*combinations(98,0) / combinations(100,2)

0.00020202020202020202

Ответ: 0.02%